source/tests/system/nisync_session_tests.cpp

```diff
@@ -0,0 +1,117 @@
+#include <gtest/gtest.h>
+
+#include "nisync/nisync_library.h"
+#include "nisync/nisync_service.h"
+
+namespace ni {
+namespace tests {
+namespace system {
+
+namespace nisync = nisync_grpc;
+
+static const int kSyncDeviceNotFound = -1074118634;
+static const char* kTestRsrcName = "";
```

Comment:
It seems like niSync_init should fail with an empty resource name is passed to it.

Reply:
Yea, I meant to comment on this and somehow didn't. For testing, I hardcoded the simulated 6674T device name in this constant and built.

Reply:
For now, how about we put the following.
```suggestion
// Update the value of 'kTestRsrcName' to the name of your NI-Sync device.
static const char* kTestRsrcName = "PXI1Slot4";
```
Long-term we should use device enumeration to discover the alias of the device on the remote system.